app.rb

```diff
@@ -95,26 +112,24 @@ def create_rental
     selected_book = gets.chomp.to_i
     puts
     puts 'Select a person from the following list by number (not id)'
-    @people_list.each_with_index do |person, index|
+    @people_list.instance_variable_get(:@people).each_with_index do |person, index|
       puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
     end
     selected_person = gets.chomp.to_i
-
     print 'Date: '
     date = gets.chomp
-    @rentals_list.push(Rental.new(date, @books_list[selected_book], @people_list[selected_person]))
+    rental = Rental.new(date, @books_list[selected_book],
+                        @people_list.instance_variable_get(:@people)[selected_person])
+    @rentals_list.add_rental(rental)
     puts 'Rental created successfully'
     @parent.show_menu
   end
 
   def list_all_rentals
     print 'ID of person: '
-    id = gets.chomp.to_i
-
+    person_id = gets.chomp.to_i
     puts 'Rentals:'
-    @rentals_list.each do |rental|
-      puts "Date: #{rental.date}, Book \"#{rental.book.title}\" by #{rental.book.author}" if rental.person.id == id
-    end
+    @rentals_list.list_all_rentals(person_id)
     @parent.show_menu
```

Comment:
Nice job so far😉. I noticed that both the main class and the app class share the responsibility of requesting and validating user inputs. To improve the code's maintainability and adhere to the Single Responsibility Principle (SRP) or SOLID principles, I would suggest you assign the responsibility of validating and requesting inputs to the `main` class😉. In implementing this, kindly take all prompts or commands to request and validate inputs to the `main` class and you can call them here where necessary😉.
By doing this, you'll achieve a clearer separation of concerns and improve the codebase's overall structure. Each class will have a single responsibility, making the code easier to understand, modify, and test.

you can take a look at [this lesson](https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design) for a better understanding on the SOLID principle😉.

Reply:
Hi @Ibizugbe ,
Thanks for reviewing our code, we will come up with a solution.